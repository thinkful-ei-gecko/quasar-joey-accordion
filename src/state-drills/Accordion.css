button{
    font-size: 22px;
}
ul{
list-style: none;
display: flex;
flex-direction: column;
justify-content: center;
align-items: center;
}
li{
margin: 100px;
display: flex;
flex-direction: column;
justify-content: center;
align-items: center;

}
li p{
background-color: rgb(32, 32, 41);
color: white;
}