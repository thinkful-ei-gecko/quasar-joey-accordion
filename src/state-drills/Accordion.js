import React from 'react';

export default class Accordion extends React.Component{
state ={
    buttonIndex: null
}
static defaultProps = {
     sections: []
    }
handleButtonClick(index){
    this.setState({buttonIndex: index})

}
render(){
    
    let sections = this.props.sections.map( (section, index) => {
        return(
            <li key={index}><button onClick={()=>{
                this.handleButtonClick(index)}}>{section.title}</button>
                {this.state.buttonIndex === index ? (<p>{section.content}</p>):''}
            </li>
        )
    })
    return(<ul>
       {sections}
    </ul> 
    )
}

}
