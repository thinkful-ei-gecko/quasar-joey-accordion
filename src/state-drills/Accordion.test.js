import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordion from './Accordion';
import { exportAllDeclaration } from '@babel/types';


describe('Accordion Component', () => {

  const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]

  //  renders an empty li when the sections prop is not supplied
  it('renders an empty li when sections prop is not supplied', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  //  renders no sections as active by default
  it('renders no sections as active by default', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  //  opens a clicked section
  it('opens a clicked section', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(0).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('only opens the last section when multiple sections have been clicked', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(0).simulate('click')
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})




//  opens a clicked section
//  only opens the last section when multiple sections have been clicked.