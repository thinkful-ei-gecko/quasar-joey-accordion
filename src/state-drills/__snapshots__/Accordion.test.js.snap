// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component only opens the last section when multiple sections have been clicked 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component opens a clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders an empty li when sections prop is not supplied 1`] = `<ul />`;

exports[`Accordion Component renders no sections as active by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
